jscomp/bsb/bsb_config_parse.ml

```diff
@@ -146,6 +145,63 @@ let extract_reason_react_jsx (map : json_map) =
   |> ignore;
   !default
 
+let extract_jsx_version (map : json_map) =
+  let default : Bsb_config_types.jsx_version option ref = ref None in
+  map
+  |? ( Bsb_build_schemas.jsx,
+       `Obj
+         (fun m ->
+           match m.?(Bsb_build_schemas.jsx_version) with
+           | Some (Flo { loc; flo }) -> (
+               match flo with
+               | "3" -> default := Some Jsx_v3
+               | "4" -> default := Some Jsx_v4
+               | _ -> Bsb_exception.errorf ~loc "Unsupported jsx-version %s" flo
+               )
+           | Some x ->
+               Bsb_exception.config_error x
+                 "Unexpected input (expect a version number) for jsx-version"
+           | None -> ()) )
+  |> ignore;
+  !default
+
+let extract_jsx_module (map : json_map) =
+  let default : Bsb_config_types.jsx_module option ref = ref None in
+  map
+  |? ( Bsb_build_schemas.jsx,
+       `Obj
+         (fun m ->
+           match m.?(Bsb_build_schemas.jsx_module) with
+           | Some (Str { loc; str }) -> (
+               match str with
+               | "react" -> default := Some React
+               | _ -> Bsb_exception.errorf ~loc "Unsupported jsx-module %s" str)
+           | Some x ->
+               Bsb_exception.config_error x
+                 "Unexpected input (jsx module name) for jsx-mode"
+           | None -> ()) )
+  |> ignore;
+  !default
+
+let extract_jsx_mode (map : json_map) =
```

Comment:
There's a bunch of repeated work here -- how about extract all 3 values at once?

Reply:
Fixed https://github.com/rescript-lang/rescript-compiler/pull/5484/commits/9cb6590478c7fdf9b9967367d0f3b294f099a957